clang/lib/Sema/SemaExpr.cpp

```diff
@@ -9116,6 +9128,23 @@ QualType Sema::GetCheckedCInteropType(ExprResult LHS) {
         IsParam = isa<ParmVarDecl>(Var);
       }
     }
+    // If `e` has bounds-safe interface type with referent type T, then
```

Comment:
I think the wording on this comment is somewhat unclear.   It's subtle to say that e has a referent type.   How about `e` has a bounds-safe interface type that is a pointer to T?